### ISSUE TITLE
Error when requesting a Mystery Seed

### ISSUE THREAD

**Issue by smithlevenson** [tcprescott/pyz3r#13]
Sorry I didn't know where else to put this so I apologize if this is the wrong spot. Awesome program BTW!
 
It took some editing of mystery.py to get the generate_random_settings() to give me a dict that can be used with seed = pyz3r.alltpr(settings=settings). Mystery.py was adding a quickswap and pot shuffle setting that didn't work with the seed generation and then "return settings, customizer" made the settings variable type a tuple instead of dict. Perhaps I am not smart enough to have converted it, but when I removed quickswap and pot suffle options and deleted the customizer return the seed generation worked perfectly. 

TL;DR The sample mystery seed generation code didn't work as shown in the Readme. I had to modify mystery.py to make it function.

**Comment by tcprescott** [comment]
I'm in the process of rewriting pyz3r.  I'll make sure the mystery functionality works as indicated in the README when I redo the documentation.  Thanks for the report!

**Comment by smithlevenson** [comment]
Hey just a quick question. Do you know why entrance_shuffle isn't in the BASE_CUSTOMIZER_PAYLOAD? I am guessing that you can't set that in the alttp.com customizer?

**Comment by cassidoxa** [comment]
That's correct. The item-only randomizer and entrance randomizer are two separate randomizers which alttpr.com provides a unified interface for but the customizer currently only provides an interface for the former.

**Comment by smithlevenson** [comment]
Oh, wow. That is interesting. I noticed that the website doesn't have a toggle for entrance randomizer on the custom game menus. So even SahasrahBot can only choose between custom item and entrance shuffle (in essence)? 
That's too bad. It would be nice to drop some items into a seed to negate some harder settings. And I am sure it would simplify the code to have one master Payload to deal with.

**Comment by smithlevenson** [comment]
I was digging around and saw some other options like "region.bossesHaveItem" that would seem to be used in things like Ambrosia, but I don't see it in the YAML. I know this isn't the same repository, but are there other settings that can be used, just not with Pyz3r?

Thanks again for all the responses. I know you guys are busy and I hate wasting your time.

**Comment by tcprescott** [comment]
`region.bossesHaveItem` comes from the randomizer website itself.  You can find these in the source code (they're not formally documented).

These are flags that are passed along with the `custom` key in the customizer payload.  The "settings" that get sent are, in the end, just serialized to json and sent to `/api/customizer` endpoint (assuming customizer is set to true).  The only reason why I have `BASE_CUSTOMIZER_PAYLOAD` is for the code that converts customizer save output to a payload that can be submitted as settings.

**Comment by tcprescott** [comment]
Absolutely!  Keep in mind I'm working on a major release that has a ton of breaking changes, so you may need to make changes down the road.

**Comment by smithlevenson** [comment]
That's fine. I am looking forward to the update!
